docs/src/nlp.md

```diff
@@ -1,249 +1,391 @@
 Nonlinear Modeling
 ==================
 
-TODO: This is out of date and has not yet been updated for MOI.
+```@meta
+CurrentModule = JuMP
+DocTestSetup = quote
+    using JuMP
+end
+```
 
-JuMP has support for general smooth nonlinear (convex and nonconvex) optimization problems. JuMP is able to provide exact, sparse second-order derivatives to solvers. This information can improve solver accuracy and performance.
+JuMP has support for general smooth nonlinear (convex and nonconvex)
+optimization problems. JuMP is able to provide exact, sparse second-order
+derivatives to solvers. This information can improve solver accuracy and
+performance.
 
-Nonlinear objectives and constraints are specified by using the `@NLobjective` and `@NLconstraint` macros. The familiar `sum()` syntax is supported within these macros, as well as `prod()` which analogously represents the product of the terms within. Note that the `@objective` and `@constraint` macros (and corresponding functions) do *not* currently support nonlinear expressions. However, a model can contain a mix of linear, quadratic, and nonlinear constraints or objective functions. Starting points may be provided by using the `start` keyword argument to `@variable`. If a starting value is not provided for a variable, it will be set to the projection of zero onto the interval defined by the variable bounds. For nonconvex problems, the returned solution is only guaranteed to be locally optimal. Convexity detection is not currently provided.
+Nonlinear objectives and constraints are specified by using the `@NLobjective`
+and `@NLconstraint` macros. The familiar `sum()` syntax is supported within
+these macros, as well as `prod()` which analogously represents the product of
+the terms within. Note that the `@objective` and `@constraint` macros (and
+corresponding functions) do *not* currently support nonlinear expressions.
+However, a model can contain a mix of linear, quadratic, and nonlinear
+contraints or objective functions. Starting points may be provided by using the
+`start` keyword argument to `@variable`.  For nonconvex problems, the returned
+solution is only guaranteed to be locally optimal. Convexity detection is not
+currently provided.
 
-For example, we can solve the classical Rosenbrock problem (with a twist) as follows:
+TODO(issue #1460): Describe how starting points are computed if none are
+provided.
+
+For example, we can solve the classical Rosenbrock problem (with a twist) as
+follows:
 
 ```julia
-using JuMP
-m = Model()
-@variable(m, x, start = 0.0)
-@variable(m, y, start = 0.0)
+using Ipopt
+model = Model(with_optimizer(Ipopt.Optimizer))
+@variable(model, x, start = 0.0)
+@variable(model, y, start = 0.0)
 
-@NLobjective(m, Min, (1-x)^2 + 100(y-x^2)^2)
+@NLobjective(model, Min, (1 - x) ^ 2 + 100 * (y - x ^ 2) ^ 2)
```

Comment:
Spacing with `^` seems odd. Doesn't the `100(` still work or do we need `100 * (` ?

Reply:
> Maybe, or maybe we're not used to the binary operator convention. Should be raised as a style question.

here the binary operator is special since it is using as a replacement for putting it in exponent so we expect it to be close, smaller and slightly moved up. We cannot really make it smaller and move it up but at least we want it to be close

For the `100(`, I also though that since it is in the doc we might want to show up that in Julia we are not forced to use `*` in some situations